server/grammar/Yml.g4

```diff
@@ -1,458 +1,482 @@
+    (
+        optionalArguments+=optionalArgDecl
+        (
+            COMMA optionalArguments+=optionalArgDecl
+        )*
+    )? CLOSE_BRACE argSetName=ymlId?
+;
+
+optionalArgDecl:
+    (
+        OPEN_BRACKET optionalKeyName=ymlId CLOSE_BRACKET
+        | mandatoryKeyName=ymlId
+    ) COLON argType=ymlId argName=ymlId? argOptionList?
+;
+
+argOptionList:
+    OPEN_BRACE arguments+=argOptionBlock (COMMA arguments+=argOptionBlock)* CLOSE_BRACE
+;
+
+argOptionBlock: argOption=ymlId optionValue=value?;
+
+order1Block: instanciationCondition conditionBlock?;
+
+instanciationCondition: inlineOperation;
+
+order1FullCondition: conditionBlock? order1Block*;
+
+//Comparisons
+combinedComparison:
+    OPEN_PAR combinedComparison CLOSE_PAR
+    | leftCondition=combinedComparison COND_AND rightCondition=combinedComparison
+    | leftCondition=combinedComparison COND_OR rightCondition=combinedComparison
+    | comparison
+;
+
+comparison: leftValue=value comparisonOperator rightValue=value;
+
+comparisonOperator:
+    EQUAL_COMP
+    | NOT_EQUALS
+    | LESS_OR_EQUAL
+    | GREATE_OR_EQUAL
+    | STRICT_LESS
+    | STRICT_GREAT
+;
+
+instruction_multivaluedAssignment:
+    leftHand=value MULTIVALUED_ASSIGNMENT rightHand=value
+;
+instruction_assignment: leftHand=value EQUAL_ASSIGNMENT rightHand=value;
+
+conditionBlock: order0Condition+;
+
+order0Condition: combinedComparison | existentialOperator;
+
+instruction_switchCase_withValue:
+    SWITCH OPEN_PAR value CLOSE_PAR OPEN_BRACE instructionCase* instructionDefault? CLOSE_BRACE
+;
+instruction_switchCase_asIf:
+    SWITCH OPEN_BRACE instructionCase* instructionDefault? CLOSE_BRACE
+;
+instructionCase:
+    CASE OPEN_PAR valueOrCondition CLOSE_PAR COLON (instruction | actionBlock)
+;
+instructionDefault: DEFAULT COLON (instruction | actionBlock);
+instruction_break: BREAK SEMICOLON?;
+
+instruction_ifElse: instruction_if (ELSE (actionBlock | instruction))?;
+instruction_if:
+    IF OPEN_PAR order0Condition CLOSE_PAR (actionBlock | instruction)
+;
+instruction_forall:
+    FORALL OPEN_PAR (ymlId | instanciationVariable) IN (value | FUNCTION) CLOSE_PAR
+    (
+        actionBlock
+        | instruction
+    )
+;
+instruction_while: WHILE OPEN_PAR order0Condition CLOSE_PAR actionBlock;
+instruction_return: RETURN value SEMICOLON?;
+instruction_chainedCall: chainedCall;
+instruction:
+    instruction_chainedCall SEMICOLON?
+    | instruction_multivaluedAssignment SEMICOLON?
+    | instruction_assignment SEMICOLON?
+    | instruction_for
+    | instruction_forEach
+    | instruction_forall
+    | instruction_return
+    | instruction_ifElse
+    | instruction_try_catch SEMICOLON?
+    | instruction_switchCase_asIf
+    | instruction_break
+    | instruction_switchCase_withValue
+    | instruction_ifExprBlock
+    | instruction_while
+;
+instruction_do: 'do' actionBlock;
+instruction_try_catch:
+    'try' OPEN_PAR instruction_do 'catch' OPEN_PAR (ymlId (COMMA ymlId)*?) CLOSE_PAR actionBlock CLOSE_PAR
```

Comment:
No allcaps-token-constant-thingies for “do”, “try” and “catch”? Because just one occurrence? It's surprising since in this file you can scroll for hours without seeing any hard-coded string and then PAM, “do try catch”.

Reply:
Woops. It seems I was lazy. Sorry. I'll fix that.